sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1258,6 +1263,27 @@ public boolean apply(@Nonnull Method input) {
         iface.getName());
   }
 
+  private static boolean checkInheritedFrom(Class<?> checkClass, Class fromClass) {
+    if (checkClass.getInterfaces().length == 0) {
+      return checkClass.equals(fromClass);
+    }
+
+    boolean valid = true;
+
+    for (Class<?> klass: checkClass.getInterfaces()){
+      valid &= checkInheritedFrom(klass, fromClass);
+    }
+
+    return valid;
+  }
+
+  private static void validateInheritedFromPipelineOptions(Class<?> klass) {
+    if (!checkInheritedFrom(klass, HasDisplayData.class)) {
```

Comment:
Shouldn't this be PipelineOptions and not HasDisplayData?

Reply:
That's right. I changed that.